src/GitHub.App/Services/PullRequestService.cs

```diff
@@ -385,6 +413,23 @@ string CreateUniqueRemoteName(IRepository repo, string name)
             return uniqueName;
         }
 
+        async Task<string> ExtractToTempFile(IRepository repo, string commitSha, string fileName, Encoding encoding)
+        {
+            var contents = await gitClient.ExtractFile(repo, commitSha, fileName) ?? string.Empty;
+            return CreateTempFile(fileName, commitSha, contents, encoding);
+        }
+
+        static string CreateTempFile(string fileName, string commitSha, string contents, Encoding encoding)
+        {
+            var tempDir = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
+            var tempFileName = $"{Path.GetFileNameWithoutExtension(fileName)}@{commitSha}{Path.GetExtension(fileName)}";
+            var tempFile = Path.Combine(tempDir, tempFileName);
+
+            Directory.CreateDirectory(tempDir);
```

Comment:
We have an OS facade to abstract the filesystem out for these things. We should use it everywhere, for consistency.